html/components/team-time-tab.js

```diff
@@ -706,10 +706,16 @@ function createTeamTable() {
 				container.children('[skater="'+k.Skater+'"]').remove();
 				if (v != null && WS.state[prefix + '.Skater('+k.Skater+').Role'] != 'NotInGame') {
 					var number = WS.state[prefix + '.Skater('+k.Skater+').Number'];
-					var button = $('<button>').attr('number', number).attr('skater', k.Skater).text(number)
+					var button = $('<button>').attr('number', number).attr('skater', k.Skater)
+							.attr('id', 'Team'+team+pos+k.Skater).addClass('KeyControl').text(number)
 							.click(function() {
-						WS.Set(prefix + '.Position('+pos+').Skater', k.Skater);
-					}).button();
+								WS.Set(prefix + '.Position('+pos+').Skater', k.Skater);
+						}).button();
+						_crgKeyControls.setupKeyControl(button, _crgKeyControls.operator);
+						var key = WS.state[button.attr('_crgKeyControls_prop')];
+						button.toggleClass('HasControlKey', (key?true:false))
```

Comment:
Why do you need this? setupKeyControl should already have us covered here.

Reply:
It didn't work in all cases. (IIRC when the button was removed and readded the key wasn't activated, while it did work when reloading the screen.) 

> I believe this will be the last UI change before v4?

I don't have anything else in the pipeline.

> I also noticed that we're not excluding the bench coach from this list.

Can add that tonight.

Reply:
I think we'd be better off fixing this over in setupKeyControl, rather than spreading the key control code around.

Reply:
I was shying away from touching this code to avoid introducing some subtle bug shortly before release. But before this turns into another tug of war...

> I also noticed that we're not excluding the bench coach from this list.

Actually, we are. Bench coach is mapped to NotInGame.

Reply:
> I was shying away from touching this code to avoid introducing some subtle bug shortly before release.

I'd be more worried about longer term maintenance, duplicating code like that is asking for one copy not to be updated down the line. No obvious issues from a quick test, and I'll be using this at a tournament and it'll get a full run through when I'm doing the videos.

> Actually, we are. Bench coach is mapped to NotInGame.

So it is, it looks like it doesn't interact too well with such a skater being Jammer/Pivot when their state is changed.